skt/executable.py

```diff
@@ -194,57 +203,61 @@ def cmd_merge(cfg):
                                         'merge.report')
 
     try:
-        if cfg.get('merge_ref'):
-            report_string = '{}\n{}'.format(
-                report_string,
-                'We merged the following references into the tree:'
-            )
-            for mb in cfg.get('merge_ref'):
-                save_state(cfg, {'mergerepo_%02d' % idx: mb[0],
-                                 'mergehead_%02d' % idx: bhead})
-                (retcode, bhead) = ktree.merge_git_ref(*mb)
-
-                utypes.append("[git]")
-                if retcode:
-                    if retcode != SKT_ERROR:
-                        report_results(merge_result_path, 'false',
-                                       merge_report_path, report_string)
-
-                    return
-
-                report_string += '  - %s\n    into commit %s' % (mb[0],
-                                                                 bhead[:12])
-                idx += 1
-
-        elif cfg.get('patch') or cfg.get('pw'):
-            report_string = 'We applied the following patch(es):\n'
-
-            if cfg.get('patch'):
-                utypes.append("[local patch]")
-                for patch in cfg.get('patch'):
-                    patch = os.path.abspath(patch)
-                    save_state(cfg, {'localpatch_%02d' % idx: patch})
-                    report_string += '  - %s\n' % patch
-
-                    ktree.merge_patch_file(patch)
-                    idx += 1
-
-            elif cfg.get('pw'):
-                utypes.append("[patchwork]")
-                for patch in cfg.get('pw'):
-                    save_state(cfg, {'patchwork_%02d' % idx: patch})
-                    report_string += '  - %s,\n' % get_patch_name(
-                        get_patch_mbox(patch)
+        if cfg.get('order_merge'):
```

Comment:
It would be easier to read if we skipped the whole try/except block if no options are passed, instead of indenting everything. Especially since we are now iterating over a single attribute.

Reply:
I do not get it `skip the whole try/except block`. Could you elaborate more? 
`Especially since we are now iterating over a single attribute.` the option `--merge-ref` still iterates the multiple attribute ( It has the for loop inside). Could you give me the example command for option `merge-ref`? Because it is not mentioned in `README`

Reply:
I mean that we are iterating over the `order_merge` attribute (can we name it something else so it doesn't sound like a command?), instead of multiple attributes. So instead of

```
try:
    if cfg.get('order_merge'):
        for task in cfg['order_merge']
            ....
```

we could have

```
for thing_to_merge in cfg.get('order_merge', []):
    try:
        ....
```

Granted, we need to make sure that the option isn't initialized to `None` but `[]` but this should be handled by the custom action.

The `merge-ref` option parsed by command line is slightly broken right now and only grabbing it from config file works (I'll take a look at fixing that later), but the format is available in the merge help. After parsing, the format of the parsed option should be a list of `[url, reference]` sublists (similarly as the `patch` is a list of patch paths).

Does this answer your questions?

Reply:
Awesome! I know what you mean.